### ISSUE TITLE
some websites entries have icons, some don't

### ISSUE THREAD

**Issue by tigrouind** [buttercup/buttercup-desktop#1071]
It seems Buttercup automatically download the favicon of a website if url is provided (which is nice).

What seems weird is that some websites get an icon, and some don't (despite proper url being set and favicon available).

Examples of websites entries that get an icon :

https://www.atlassian.com/
https://socialclub.rockstargames.com/

**Comment by perry-mitchell** [comment]
Yeah, this is an issue with our icon scraper. Some sites fail when parsing icons. Unfortunately the scraper is not currently open source, and would need my time to debug it. It's been on my to-do list for a long while but I haven't yet had a chance to attack the problem. I'll try to get to it soon.

**Comment by julianpoemp** [comment]
When I open the DevTools I see a lot of 404-errors because the icons can't be fetched from buttercup. I can understand that this solution should avoid icons to be fetched directly from target webpages. But that there are two points:

1) If users don't want to leave any traces in their network traffic they don't want to fetch images with the domains as their names. For example if `www.example.com` would be a webpage I don't want to be connected with and buttercup does an request to `https://icon.buttercup.pw/icon/www.example.de`. So if someone monitors my network traffic he would know that I have an account on this page.

2) I'm 100% sure that you don't do anything bad with the information and you don't analyze which web accounts IP address X is using. But hypothetical if I were very safety-conscious I wouldn't want you to know which web pages I'm using. Even if you say that you don't do something with it.

I know it should be as user friendly as possible but perhaps there are other solutions that do not leak any information about the webpages saved in a vault:

a) Still user-friendly, but less privacy aware: Rename images with hashes instead of the domain name: E.g. instead of `https://icon.buttercup.pw/icon/login.live.com` a request with `https://icon.buttercup.pw/icon/e7afc47ba16258b69105b2b453a3ff7a35d6d39c` would be made. In this example I created an sha1 hash of `login.live.com`.

b) Less user-friendly but more privacy aware: The icon fetching is optional and can be enabled on the settings page. Users can upload their own images, or fetch it with one click on a "fetch icon" button.

c) A combination of a) and b): Icon fetching is optional but if enabled the file name of an icon is hashed. If enabled and there is no icon the user can select its own one. If disabled the user can still set icons manually.

**Comment by tigrouind** [comment]
@julianpoemp since those requests are encrypted trough HTTPS, even if someone would analyze network traffic, I think all he would see are requests to icon.buttercup.pw server (but not the actual request content).

Additionally : Buttercup should cache those icons internally (eg: save them somewhere locally). I don't see the point to fetch them every time. If someone think it could create an issue regarding security and privacy, then user should be able to turn caching on or off trough the options (with some warning). Same about going trough the icon.buttercup.pw proxy, why not have an option to bypass it ?
